README.md

```diff
@@ -1,83 +1,92 @@
-<br>
-1. SSH to the the module's computer
-    * Core1: `ssh clucky@192.168.1.69`
-    * Core2: `ssh clucky@192.168.1.70`
-    * Arm: `ssh arm@192.168.1.70`
-    * Password: \<can be found in the rover-Docs repo under documentation>
-2. Run the  headless node
-    * You must have ANCHOR or the module's Standalone node running
-    * Open a new terminal (SSH'd to the module)
-    * Navigate to rover-ros2 workspace
-        * `cd rover-ros2`
-    * Source the workspace
-        * `source install/setup.bash`
-    * Run the node (ensure controller is connected and on x-input mode)
-        * CORE: `ros2 run core_pkg headless`
-        * ARM: `ros2 run arm_pkg headless`
+## About
+
+Includes all ROS2 packages for the rover. These are centrally located for modular rover operation.
+
+You will use this package to launch all module-side ROS2 nodes.
+
+# Table of Contents
+
+1. Title
+3. Table of Contents
+3. Software Requirements
+    1. ROS2 Humble
+    2. IP Configuration
+4. Hardware Requirements
+5. Recommended Programs
+6. How to Use
+7. Common Problems/Troubleshooting
+8. Major To-Do Items
+9. Author(s) 
+10. Maintainer(s)
+
+# Software Requirements 
+
+Since July/August of 2025, ASTRA uses NixOS. A functioning NixOS installation is assumed.
+
+## ROS2 Humble
+
+We use [ROS2 Humble](https://docs.ros.org/en/humble/index.html), and plan on using that until support for it expires. 
+
+To install ROS2, run the following command in your /etc/nixos folder:
+```nix
+nix flake init --template github:lopsided98/nix-ros-overlay
+```
+
+After you install ROS2, you can install all dependencies needed by this repository by running the following commands:
+
+```bash
+$ sudo rosdep init  # only run once
+$ rosdep update
+$ rosdep install --from-path . --ignore-src -y
+```
+
+## IP Configuration
+
+- IP Address: 192.168.1.x
+    - This can be just about anything not already in use. I recommend something 30-39
+- Net Mask: 255.255.255.0
+- Gateway: 192.168.1.0
+
+# Hardware Requirements 
+
+- A functioning computer.
+
+# Recommended Programs
+
+## VSCode
+
+VSCode is a wonderful program. I used it to make this project and recommend anyone else working on it use it as well. 
+
+To install VSCode, add the following package to your pkgs array / packages.nix config file:
+```nix
+vscode-fhs
+```
+Alternatively, use a flake to install home-manager, and add to your home.nix:
+```nix
+vscode.enable = true;
+```
+
+# How to Use
+
+```bash
+$ colcon build
+$ source install/setup.bash
+  # main launch files:
+$ ros2 launch anchor_pkg rover.launch.py
+$ ros2 run headless_pkg headless_full
+```
+
+## Testing Serial
+
+You can fake the presence of a Serial device (i.e., MCU) by using the following command:
+
+```bash
+socat -dd -v pty,rawer,crnl,link=/tmp/ttyACM9 pty,rawer,crnl,link=/tmp/ttyOUT
```

Comment:
should also mention the environment variable you must pass into anchor: PORT_OVERRIDE